core/trino-main/src/main/java/io/trino/execution/scheduler/SqlQueryScheduler.java

```diff
@@ -1006,6 +1018,20 @@ public void onTaskFailed(TaskId taskId, Throwable failure)
         }
     }
 
+    /**
+     * Scheduler for stages that are not coordinator only
```

Comment:
`workers only`

`not coordinator only` suggest they might run on coordinator and workers

Reply:
In theory (if scheduling on coordinator is allowed) some stage may have some tasks scheduled on the coordinator

Reply:
> some stage may have some tasks scheduled on the coordinator

Yes, but logically that would be part of "worker", no? Anyways, scheduling on coordinator is off in production env.